api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -364,6 +425,11 @@ type ManagedControlPlaneSubnet struct {
 
 // AzureManagedControlPlaneStatus defines the observed state of AzureManagedControlPlane.
 type AzureManagedControlPlaneStatus struct {
+	// AutoUpgradeVersion is the Kubernetes version populated after autoupgrade based on the upgrade channel.
+	// +kubebuilder:validation:MinLength:=2
```

Comment:
```suggestion
	// +kubebuilder:validation:MinLength=2
```
I think it won't work with a semicolon btw

Reply:
may be, not sure the same was used for spec.version.